### DISCUSSION TITLE
Limit search for high-precision coefficients and constants?

### DISCUSSION THREAD

**Discussion by fishwithfros** [MilesCranmer/PySR#244] [General]
I have a particular PySR project that seems to spend lots of time trying to dial in levels of precision on coefficients and constants that I just don't need. For this project, I could easily live with two significant digits.

Is there a way to reduce the precision of the coefficients and constants, and would it help the search process at all?

**Comment by MilesCranmer** [comment]
You can reduce the frequency of constant optimizations with `optimize_probability` or even turn it off altogether with `should_optimize_constants=False`. That will give you a decent speed up.

You can also play around with the `mutate_constant` parameter, which lets the evolutionary algorithm tweak the constants.

There’s also the `precision` parameter, which you could try lowering to 16, although it probably depends on your CPU whether this actually gives a speed up (for me, it was actually slower).

Cheers,
Miles
